core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -104,12 +94,20 @@ object TestUtils extends Logging {
    * Create a temporary relative directory
    */
   def tempRelativeDir(parent: String): File = {
-    val f = new File(parent, "kafka-" + random.nextInt(1000000))
-    f.mkdirs()
+    new File(parent).mkdirs()
+    val f: File = Iterator.continually(new File(parent, "kafka-" + random.nextInt(1000000))).find(_.mkdir()).get
```

Comment:
I just realised that we could end up in an infinite loop if `new File(parent).mkdirs()` fails. Should we check for that?